lib/hashie/extensions/coercion/active_model.rb

```diff
@@ -0,0 +1,85 @@
+require 'active_model/type'
+
+module Hashie
+  module Extensions
+    module Coercion
+      module ActiveModel
+        # Symbol is the one glaring type omission. Define it if it is not
+        # already defined.
+        begin
+          ::ActiveModel::Type.lookup(:symbol)
+        rescue ArgumentError
+          class Symbol < ::ActiveModel::Type::String
+            def type
+              :symbol
+            end
+
+            private
+
+            # The symbol type leverages in coercion logic in the String
+            # superclass and simply uses `to_sym` to covert to a symbol.
+            def cast_value(value)
+              super.to_sym
+            end
+          end
+          ::ActiveModel::Type.register(:symbol, Symbol)
+        end
+
+        # Array of symbols of the included ActiveModel types.
+        ACTIVE_MODEL_TYPES = ::ActiveModel::Type.registry
+                             .send(:registrations)
+                             .map { |r| r.send(:name) }
+                             .freeze
+
+        def build_coercion(type)
```

Comment:
This is a very long method. I think it would help understanding to break it apart at each case:

```ruby
def build_coercion(type)
  if type.is_a? Enumerable
    build_enumerable_coercion(type)
  elsif ACTIVE_MODULE_TYPES.include? type
    build_active_model_coercion(type)
  # ...
  end
end
```

Ideally, I would like to see this class broken up a little more (i.e. into single responsibility objects that know how to create a coercion of each type), but we can work on that after we hammer everything else out.